atlasdb-dbkvs-hikari/src/main/java/com/palantir/nexus/db/pool/HikariCPConnectionManager.java

```diff
@@ -354,9 +359,17 @@ private HikariDataSource getDataSourcePool() {
                 }
             }
         } catch (PoolInitializationException e) {
+            // Intentionally ignoring suppressed exceptions
+            // Keep duplicates to safeguard the full causality chain
+            List<Integer> vendorSqlErrorCodes = Streams.stream(
+                            Iterables.filter(Throwables.getCausalChain(e), SQLException.class))
```

Comment:
nit: mixing streams with Guava lazy filter seems odd, might want to just stream:

```suggestion
            List<Integer> vendorSqlErrorCodes = Throwables.getCausalChain(e).stream()
                    .filter(SQLException.class::isInstance)
                    .map(SQLException.class::cast)
```